tests/engine_test.py

```
@@ -35,6 +35,43 @@
+    """Build a clone shaped like a CI checkout of a pull request.
+
+    One commit on origin/main, one commit of work on top, then every local
+    branch removed and HEAD detached — so the target exists only as a
+    remote-tracking ref and the branch name resolves to nothing.
+
+    Returns the clone path.
+    """
+    import subprocess as sp
+
+    def git(*args, cwd):
+        return sp.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)